developer/src/kmc-kmn/test/test-messages.ts

```diff
@@ -112,4 +112,20 @@ describe('KmnCompilerMessages', function () {
     await testForMessage(this, ['invalid-keyboards', 'error_character_range_too_long.kmn'], KmnCompilerMessages.ERROR_CharacterRangeTooLong);
   });
 
+  // WARN_IndexStoreShort
+
+  it('should generate WARN_IndexStoreShort if a store referenced in index() is shorter than the corresponding any() store', async function() {
```

Comment:
I see `WARN_IndexStoreShort` already exists. 

Why is _Short a WARN and _Long a HINT?

Reply:
Short is a warning because it has negative side-effects because inputs may result in bogus outputs. Long is just a hint because it is perfectly valid, but may be a sign that there is an unintentional mismatch.